packages/node/src/indexer/fetch.service.ts

```diff
@@ -435,14 +441,18 @@ export class FetchService implements OnApplicationShutdown {
           this.eventEmitter.emit(IndexerEvent.SkipDictionary);
         }
       }
-      // the original method: fill next batch size of blocks
       const endHeight = this.nextEndBlockHeight(
         startBlockHeight,
         scaledBatchSize,
       );
-      this.blockDispatcher.enqueueBlocks(
-        range(startBlockHeight, endHeight + 1),
-      );
+
+      if (this.getModulos().length > 0) {
+        this.runModuloBlocks(startBlockHeight, endHeight);
```

Comment:
End height here wouldn't be right. The modulo is only going to be a subset of this and we might not fill the buffer. 

e.g. if modulo is 100 and batch size is 10 we would have 0 blocks added